### DISCUSSION TITLE
How to specify the finality

### DISCUSSION THREAD

**Discussion by koa202405** [wevm/wagmi#4149] [Q&A] [answered]
When I use `writeContract` I found that it will return before execution finished, how can I make it only return after execution?

**Comment by jxom** [comment, marked as answer]
`writeContract` returns when the transaction has been broadcast. You can utilize [`waitForTransactionReceipt`](https://wagmi.sh/core/api/actions/waitForTransactionReceipt) to check if the transaction has been included on a Block (ie. executed).

**Reply by koa202405** [reply]
Thanks!
